File: state/typescript/delivery-state.ts
import {VendingMachineState} from "./vending-machine-state";
import {VendingMachine} from "./vending-machine";
import {WaitingState} from "./waiting-state";

// Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
export class DeliveryState extends VendingMachineState {
  // DeliveryState durumunda otomatadan ürünün teslim işlemleri yapılmaktadır.
  // Ürün verildikten sonra başlangıç, bekleme durumuna geçmektedir. (WaitingState)
  public handleState(context: VendingMachine): void {
    console.log('DeliveryState: Ürün teslim ediliyor.');

    context.setState(new WaitingState());
  }
}

File: state/go/delivery-state.go
package main

import "fmt"

// deliveryState Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
type deliveryState struct{}

// deliveryState durumunda otomatadan ürünün teslim işlemleri yapılmaktadır.
// Ürün verildikten sonra başlangıç, bekleme durumuna geçmektedir. (waitingState)

func NewDeliveryState() VendingMachineState {
	return deliveryState{}
}

func (s deliveryState) HandleState(context *VendingMachine) {
	fmt.Println("deliveryState: Ürün teslim ediliyor.")
	context.SetState(NewWaitingState())
}
